### PR TITLE
Add 'not equal', 'less than', and 'greater than' recognition operator

### PR DESCRIPTION
**Pull request by timgerstel**
This pull request implements the following feature:
- Implement 'not equal' recognition operator using the operator "NE"
- Implement 'less than' recognition operator using the operator "LT"
- Implement 'greater than' recognition operator using the operator "GT"
- [Allow nested properties in recognizers syntax](https://github.com/zowe/zlux-platform/pull/69) @lchudinov 
```
"recognizers":[
    {
      "id":"org.zowe.zlux.sample",
      "clause": {
        "op": "AND",
        "args": [
         {"prop":["NE","foo","bar"]},{"prop":["NE","foobar","barfoo"]}
        ]
      }
    }
  ]
```

Signed-off-by: Timothy Gerstel <tgerstel@rocketsoftware.com>

### CODE REVIEW

**Review by lchudinov — COMMENTED** [review]
You implemented binary relational operator `!=`(not equal), but named it `NOT`, which is reserved for boolean unary operator Logical NOT. It works good. To implement it properly please introduce a new binary relational operator `!=` (not equal):
* Add a new member  `PROPERTY_NE` into enum `RecognitionOp`
* The new operator should be named `NE` as string

**Comment by timgerstel** [comment]
> 
> 
> You implemented binary relational operator `!=`(not equal), but named it `NOT`, which is reserved for boolean unary operator Logical NOT. It works good. To implement it properly please introduce a new binary relational operator `!=` (not equal):
> 
>     * Add a new member  `PROPERTY_NE` into enum `RecognitionOp`
> 
>     * The new operator should be named `NE` as string

I have named the new member NE, as it will make it easier in the future. This line here:
https://github.com/zowe/zlux-platform/pull/70/files#diff-2b402e0f54e51c97775217ab57dab5581419b1de12b823717d66aa2a176214c5R1112
Was the easiest way I could think of to initialize the operator based on the first array value.

File: base/src/dispatcher/dispatcher.ts [hunk 1]
```diff
@@ -1053,13 +1054,14 @@ export enum RecognitionOp {
   OR,
   NOT,
```
Comment (1000TurquoisePogs) [inline base/src/dispatcher/dispatcher.ts:1055]: but you already had NOT, how is this different from PROPERTY NE?

**Comment by 1000TurquoisePogs** [comment]
Otherwise, makes sense.

File: base/src/dispatcher/dispatcher.ts (hunk 1 above)
Comment (timgerstel) [reply base/src/dispatcher/dispatcher.ts:1055]: NOT isnt actually implemented currently, but I see not as:
A & !B -> 'A' AND 'NOT B'

where as NE is
A != B -> 'A' NOT EQUAL 'B'

NOT is unary but NE is binary

**Review by lchudinov — APPROVED** [review]
Looks and works good to me.

File: base/src/dispatcher/dispatcher.ts [hunk 2]
```diff
@@ -361,7 +361,8 @@ export class Dispatcher implements ZLUX.Dispatcher {
      this.recognizers.push(recognitionRule);
      if (predicate.operation == RecognitionOp.AND){
        for (let subClause of predicate.subClauses){
-         if ((subClause as RecognitionClause).operation == RecognitionOp.PROPERTY_EQ){
+         const operation = (subClause as RecognitionClause).operation;
+         if (operation === RecognitionOp.PROPERTY_EQ || operation === RecognitionOp.PROPERTY_NE){
```
Comment (lchudinov) [inline base/src/dispatcher/dispatcher.ts:365]: The condition must include all new operators.

**Review by lchudinov — APPROVED** [review]

**Review by 1000TurquoisePogs — APPROVED** [review]
looks good to me now.

**Pull request merged**